src/functional_data.Rmd

```diff
@@ -0,0 +1,237 @@
+
+* **Use a [learner](&Learner.md) suitable for functional data**
+
+  + For regression:
+  
+```{r}
+  # The following learners can be used for the task.
+  listLearners(tsk1, properties = "functionals")
+  # Create a FDboost learner
+  fdalrn = makeLearner("regr.FDboost")
+```
+  
+  + Or alternatively for classification:
+  
+```{r}
+  # knn learner
+  knn.lrn = makeLearner("classif.fdausc.knn")
+```
+
+* **Use a _standard_ [learner](&Learner.md)**: 
+In this case the temporal structure is disregarded, and the functional data treated as simple numeric features.
+
+```{r}
+## Decision Tree learner
+rpartlrn = makeLearner("classif.rpart")
+```
+
+* Alternatively, transform the functional data into a non-temporal/non-functional space by [extracting](functional_data.Rmd#feature-extraction) features before training.
+In this case, a normal regression- or classification-[learner](&Learner.md) 
+can be applied.
+
+This is explained in more detail in the [feature extraction](functional_data.Rmd#feature-extraction)
+section below.
+
+
+## Train the learner
+
+The resulting learner can now be trained on the task created in section [Creating a Task](functional_data.Rmd#creating-a-task) above. 
+
+```{r, warning = FALSE}
+# Train the fdalrn on the constructed task
+m = train(learner = fdalrn, task = tsk1)
+p = predict(m, tsk1)
+performance(p, rmse)
+```
+
+```{r}
+# Alternatively, we can simply resample (3-fold Cross-Validation)
+resample(fdalrn, tsk1, resampling = cv3, measures =  mse, show.info = FALSE)
+```
+
+Alternatively, learners that do not specifically treat functional covariates can
+be applied. In this case the temporal structure is completely disregarded, and all
+columns are treated as independent.
+
+```{r}
+# Train a normal learner on the constructed task.
+# Note that we get a message, that functionals have been converted to numerics.
+rpart.lrn = makeLearner("regr.rpart")
+m = train(learner = rpart.lrn, task = tsk1)
+m
+```
+
+## Feature Extraction
+
+In contrast to applying a learner that works on a [Task](&makeTask) containing functional features,
+the [Task](&makeTask) can be converted to a normal [&Task.md].
+This works by transforming the functional features into a 
+non-functional domain, e.g by extracting wavelets.
+
+The currently supported preprocessing functions are:
+* discrete wavelet transform
+* fast fourier transform
+* functional principal component analysis
+* multi-resolution feature extraction
+
+In order to do this, we specify methods for each functional feature in the task in a __list__.
+In this case we simply want to extract the mean from each __UVVIS__ functional and the fourier transformed features from each __NIR__ functional. Additional arguments can be passed on 
+
+```{r}
+# feat.methods specifies what to extract from which functional
+# from the first functional we extract the fourier transformation, from the second the fpca scores
+feat.methods = list("UVVIS" = extractFDAFourier(), "NIR" = extractFDAFPCA())
```

Comment:
Can I get both the Fourier and the fpca scores for UVVIS?

Reply:
Yes, trying to make that clearer.